File: src/main/java/com/ardaatay/issuemanagment/service/IssueHistoryService.java
package com.ardaatay.issuemanagment.service;

import com.ardaatay.issuemanagment.entity.Issue;
import com.ardaatay.issuemanagment.entity.IssueHistory;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;


public interface IssueHistoryService {

    IssueHistory save(IssueHistory issueHistory);

    IssueHistory getById(Long id);

    Page<IssueHistory> getAllPageable(Pageable pageable);

    Boolean delete(IssueHistory issueHistory);
}

File: src/webui/src/app/services/shared/issue.history.service.ts
import {Injectable} from "@angular/core";
import {ApiService} from "../api.service";
import {Observable} from "rxjs";
import {map} from "rxjs/internal/operators";

@Injectable({
  providedIn: 'root'
})
export class IssueHistoryService {

  private ISSUE_HISTORY_PATH = "/issue/history";

  constructor(private apiService: ApiService) {
  }

  getAll(page): Observable<any> {
    return this.apiService.get(this.ISSUE_HISTORY_PATH + '/pagination', page).pipe(map(
      res => {
        if (res) {
          return res;
        } else {
          return {};
        }
      }
    ));
  }

  getById(id): Observable<any> {
    return this.apiService.get(this.ISSUE_HISTORY_PATH, id).pipe(map(
      res => {
        if (res) {
          return res;
        } else {
          return {};
        }
      }
    ));
  }

  createIssueHistory(issueHistory): Observable<any> {
    return this.apiService.post(this.ISSUE_HISTORY_PATH, issueHistory).pipe(map(
      res => {
        if (res) {
          return res;
        } else {
          return {};
        }
      }
    ));
  }

  deleteIssueHistory(id): Observable<any> {
    return this.apiService.delete(this.ISSUE_HISTORY_PATH + '/' + id).pipe(map(
      res => {
        if (res) {
          return res;
        } else {
          return {};
        }
      }
    ));
  }
}
